File: Andrei/Small00/C++/lexical/SymbolTable.cpp
#include "SymbolTable.h"

SymbolTable::SymbolTable() {
    // TIP: To add the tokens.
    // m_st["???"] = ???;

    // symbols

    // keywords

    // operators

}

SymbolTable::~SymbolTable() {
}

bool SymbolTable::contains(std::string token) {
    return m_st.find(token) != m_st.end();
}

enum TokenType SymbolTable::find(std::string token) {
    return this->contains(token) ?
              m_st[token] : INVALID_TOKEN;
}

File: Andrei/Small00/Java/lexical/SymbolTable.java
package lexical;

import java.util.Map;
import java.util.HashMap;

class SymbolTable {

    private Map<String, TokenType> st;

    public SymbolTable() {
        st = new HashMap<String, TokenType>();
        
        // symbols

        // keywords

        // operators
    }

    public boolean contains(String token) {
        return st.containsKey(token);
    }

    public TokenType find(String token) {
        return this.contains(token) ?
            st.get(token) : TokenType.INVALID_TOKEN;
    }
}
